client/src/main/java/com/netflix/conductor/client/http/ClientBase.java

```diff
@@ -189,15 +197,53 @@ private Builder getWebResourceBuilder(URI URI, Object entity) {
         return client.resource(URI).type(MediaType.APPLICATION_JSON).entity(entity).accept(MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON);
     }
 
-    private void handleException(URI uri, RuntimeException e) {
-        if (e instanceof ClientHandlerException) {
-            logger.error("Unable to invoke Conductor API with uri: {}, failure to process request or response", uri, e);
-        } else if (e instanceof UniformInterfaceException) {
+    private void handleClientHandlerException(ClientHandlerException exception, URI uri){
+        String errorMessage = String.format("Unable to invoke Conductor API with uri: %s, failure to process request or response", uri);
+        logger.error(errorMessage);
+        throw new ConductorClientException(errorMessage, exception);
+    }
+
+    private void handleRuntimeException(RuntimeException exception, URI uri) {
+        String errorMessage = String.format("Unable to invoke Conductor API with uri: %s, runtime exception occurred", uri);
+        logger.error(errorMessage);
+        throw new ConductorClientException(errorMessage, exception);
+    }
+
+    private void handleUniformInterfaceException(ClientResponse clientResponse, URI uri) {
+        if (clientResponse == null) {
+            throw new ConductorClientException(String.format("Unable to invoke Conductor API with uri: %s", uri));
+        }
+        try {
+            if (clientResponse.getStatus() < 300) {
+                return;
+            }
+            String errorMessage = clientResponse.getEntity(String.class);
+            ErrorResponse errorResponse = null;
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                errorResponse = mapper.readValue(errorMessage, ErrorResponse.class);
+            } catch (IOException e) {
+                throw new ConductorClientException(clientResponse.getStatus(), errorMessage);
+            }
+            throw new ConductorClientException(clientResponse.getStatus(), errorResponse);
+        } catch (ConductorClientException e) {
+            throw e;
+        } catch (ClientHandlerException e) {
+            handleClientHandlerException(e, uri);
+        } catch (RuntimeException e) {
+            handleRuntimeException(e, uri);
+        }
+    }
+
+    private void parseException(URI uri, RuntimeException e) {
+        if (e instanceof UniformInterfaceException) {
             logger.error("Unable to invoke Conductor API with uri: {}, unexpected response from server: {}", uri, clientResponseToString(((UniformInterfaceException) e).getResponse()), e);
```

Comment:
move the logging into handleUniformInterfaceException, like handleClientHandlerException and handleRuntimeException.